src/network/networkctl.c

```diff
@@ -1038,6 +1038,48 @@ static int link_lldp_status(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
+static int link_delete_send_message(sd_netlink *rtnl, int index) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL, *reply = NULL;
+        int r;
+
+        assert(rtnl);
+
+        r = sd_rtnl_message_new_link(rtnl, &req, RTM_DELLINK, index);
+        if (r < 0)
+                return rtnl_log_create_error(r);
+
+        r = sd_netlink_call(rtnl, req, 0, &reply);
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+static int link_delete(int argc, char *argv[], void *userdata) {
+        _cleanup_(sd_netlink_unrefp) sd_netlink *rtnl = NULL;
+        int index;
+        int r, i;
+
+        r = sd_netlink_open(&rtnl);
+        if (r < 0)
+                return log_error_errno(r, "Failed to connect to netlink: %m");
+
+        for (i = 1; i < argc; i++) {
+                index = (int) if_nametoindex(*(argv + i));
+                if (index <= 0) {
+                        r = -errno;
+                        log_error_errno(r, "Failed to resolve interface %s", *(argv + i));
+                        continue;
+                }
+
+                r = link_delete_send_message(rtnl, index);
```

Comment:
Previous error will be ignored...

Reply:
We just log it and continue with other arguments.

Reply:
I think the command must fail. `ip link del` fails is the interface name is not given.

It'd be even nicer to resolve all the interface names before sending any messages. This will avoid starting the operation if we know it will fail anyway, making it more "atomic". (It won't be fully atomic of course, because something else can delete/create interfaces in parallel, but as long as no other process is changing the configuration, it will make our operation more consistent.)